### PR TITLE
docker: Build and run as a non-privileged user

### PR DESCRIPTION
**Pull request by heliocastro**
- Shell in login profile is not more needed
- Container runs always on the defined build user ( default: ort )
- Added an auxiliary script for apt proxy

Signed-off-by: Helio Chissini de Castro <heliocastro@gmail.com>

Please ensure that your pull request adheres to our [contribution guidelines](https://github.com/oss-review-toolkit/.github/blob/main/CONTRIBUTING.md). Thank you!

### CODE REVIEW

**Comment by nnobelis** [comment]
Could you please break the changes to the `Dockerfile` across several commits ? This is really hard to review.

File: Dockerfile [hunk 1]
```diff
@@ -246,115 +290,163 @@ RUN curl -Os https://dl.google.com/android/repository/commandlinetools-linux-${A
...
+ENV DART_SDK=/opt/dart-sdk
+ENV PATH=$PATH:${DART_SDK}/bin
+
+RUN --mount=type=tmpfs,target=/dart \
+    arch=$(arch | sed s/aarch64/arm64/ | sed s/x86_64/x64/) \
+    && curl -o /dart/dart.zip -L https://storage.googleapis.com/dart-archive/channels/stable/release/${DART_VERSION}/sdk/dartsdk-linux-${arch}-release.zip \
+    && unzip /dart/dart.zip
+
+#------------------------------------------------------------------------
+#  Dart
```
Comment (tsteenbe) [inline Dockerfile:310]: This should not be dart but sbt I guess

**Review by nicorikken — APPROVED** [review]
I did a detailed review of the Dockerfile. I think the current setup is way better. Also hardcoded home dirs of the various tools will make it more predictable.

**Comment by sschuberth** [comment]
> I did a detailed review of the Dockerfile. I think the current setup is way better. Also hardcoded home dirs of the various tools will make it more predictable.

While the changes in sum may be good, it's still all done in one bulk commit. Please split this into logical steps / semantically atomic commits to separate independent changes from another like @nnobelis was already asking for [above](https://github.com/oss-review-toolkit/ort/pull/6053#issuecomment-1308297973).

**Comment by heliocastro** [comment]
Superseeded

**Comment by sschuberth** [comment]
> Superseeded

Mind adding a link with what this is being superseded, for reference?

**Pull request closed without merging**